config/initializers/flipper.rb

```diff
@@ -42,10 +42,13 @@
 Flipper::UI.configuration.feature_creation_enabled = false
 # Make sure that each feature we reference in code is present in the UI, as long as we have a Database already
 FLIPPER_FEATURE_CONFIG['features'].each_key do |feature|
-  unless Flipper.exist?(feature)
-    Flipper.add(feature)
-    # default features to enabled for development and test only
-    Flipper.enable(feature) if Rails.env.development? || Rails.env.test?
+  feature_config = FLIPPER_FEATURE_CONFIG['features'][feature]
+
+  Flipper.add(feature) unless Flipper.exist?(feature)
+
+  # default features to be enabled in development and test unless override set in config/features.yml
+  if feature_config.dig('default_enabled_environments', Rails.env) && Rails.env.in?('development', 'test')
```

Comment:
Or the key could be more explicit, like `dev_disabled_by_default` and it could be a bit more like the original method:
```
Flipper.enable(feature) if (Rails.env.development? && !feature_config['dev_disabled_by_default']) || Rails.env.test?
```